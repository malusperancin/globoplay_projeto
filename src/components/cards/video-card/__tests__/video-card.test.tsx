/* eslint-disable testing-library/prefer-screen-queries */
/* eslint-disable testing-library/no-node-access */
import { render } from "@testing-library/react";
import VideoCardType from "../../video-card/video-card-type";
import VideoCard from "../video-card";

describe("VideoCard", () => {
  const card: VideoCardType = {
    UI_number: 0,
    availableFor: "teste",
    backgroundUrl: "teste",
    contentRating: "teste",
    contentRatingCriteria: [],
    description: "teste",
    duration: 20,
    episodeNumber: 21,
    formattedDuration: "teste",
    formattedRemainingTime: "teste",
    genresNames: [],
    headline: "teste",
    id: "teste",
    kind: "teste",
    liveThumbnail: "teste",
    logoUrl: "teste",
    posterUrl: "teste",
    seasonNumber: 3,
    selfRatedContent: true,
    technicalSpecs: {
      resolutions: [],
    },
    title: "teste",
    titleId: "teste",
    watchedPercent: 20,
    watchedProgress: 20,
  };

  const videoComponent = (
    <VideoCard data={card} isFocus={true} translate={100} />
  );

  const videoComponentNotFocus = (
    <VideoCard data={card} isFocus={false} translate={100} />
  );

  it("should render without throwing an error", () => {
    const { container } = render(videoComponent);

    expect(container).toMatchSnapshot();
    expect(container).toBeTruthy();
  });

  it("should be focus", () => {
    const { getByTestId } = render(videoComponent);
    const focus =
      getByTestId("video-card").getElementsByClassName("imageFocus")[0];

    expect(focus).toBeTruthy();
  });

  it("should not be focus", () => {
    const { getByTestId } = render(videoComponentNotFocus);
    const focus = getByTestId("video-card").getElementsByClassName("image")[0];

    expect(focus).toBeTruthy();
  });
});
