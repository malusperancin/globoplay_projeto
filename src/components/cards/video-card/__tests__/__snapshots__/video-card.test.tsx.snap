// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoCard should render without throwing an error 1`] = `
<div>
  <div
    class="container"
    data-testid="video-card"
  >
    <button
      class="button"
      style="transform: translate(100px);"
      type="button"
    >
      <div
        class="poster"
      >
        <img
          alt=""
          class="gradientFocus"
          src="gradient.png"
        />
        <div
          class="progressBar"
        >
          <span
            class="completed"
            style="width: 20%;"
          />
        </div>
        <span
          class="spanTime"
        >
          <p
            class="time"
          >
             
            teste
             
          </p>
        </span>
        <img
          alt=""
          class="imageFocus"
          src="teste"
        />
      </div>
      <div
        class="info"
      >
        <p
          class="title"
        >
           
          teste
           
        </p>
        <p
          class="description"
        >
           
          teste
           
        </p>
      </div>
    </button>
  </div>
</div>
`;
